sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/FileAsyncClient.java

```diff
@@ -1236,4 +1237,23 @@ public String getAccountName() {
         FileMetadataInfo fileMetadataInfo = new FileMetadataInfo(eTag, isServerEncrypted);
         return new SimpleResponse<>(response, fileMetadataInfo);
     }
+
+    /**
+     * Verifies that the file permission and file permission key are not both set and if the file permission is set,
+     * the file permission is of valid length.
+     * @param filePermission The file permission.
+     * @param filePermissionKey The file permission key.
+     * @throws IllegalArgumentException for invalid file permission or file permission keys.
+     */
+    private void validateFilePermissionAndKey(String filePermission, String  filePermissionKey) {
+        if (filePermission != null && filePermissionKey != null) {
+            throw logger.logExceptionAsError(new IllegalArgumentException(
+                FileConstants.MessageConstants.FILE_PERMISSION_FILE_PERMISSION_KEY_INVALID));
+        }
+
+        if (filePermission != null) {
+            Utility.assertInBounds("filePermission",
+                filePermission.getBytes(StandardCharsets.UTF_8).length, 0, 8 * Constants.KB);
```

Comment:
The `assertInBounds` is throwing an exception https://github.com/Azure/azure-sdk-for-java/blob/fec0fa1d35e7b13654815d927a338746e7582b9a/sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/Utility.java#L300 but we're not logging it

Reply:
may be we can pass logger to assertInBounds and let it use to log err, this way we use a client logger attached to the correct class it's gets called from

Reply:
@sima-zhu since this is not a release blocker we can track this separately if you think so.